src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -383,7 +383,17 @@ private BoundExpression ConvertConcatExprToStringIfPossible(CSharpSyntaxNode syn
                         if (ConcatExprCanBeOptimizedWithToString(operand.Type))
                         {
                             var toString = GetSpecialTypeMethod(syntax, SpecialMember.System_Object__ToString);
-                            return BoundCall.Synthesized(syntax, operand, toString);
+
+                            var type = (NamedTypeSymbol)operand.Type;
+                            var toStringMembers = type.GetMembers(toString.Name);
+                            foreach(var member in toStringMembers)
+                            {
+                                var toStringMethod = member as MethodSymbol;
+                                if (toStringMethod.GetLeastOverriddenMethod(type) == (object)toString)
```

Comment:
This optimization would cause it to be a binary breaking change to remove a `ToString` override. We don't want that. Can you please restrict it to "special" types?